xarray/core/datatree.py

```diff
@@ -480,6 +489,7 @@ def to_dataset(self) -> Dataset:
         --------
         DataTree.ds
         """
+        # TODO: copy these container objects?
```

Comment:
Note: We definitely need tests to verify the `to_dataset()` creates an new copy of the node's data. (I suspect that current it does not.)